libr/syscall/syscall.c

```diff
@@ -113,17 +151,28 @@ R_API bool r_syscall_setup(RSyscall *s, const char *arch, int bits, const char *
 		}
 	}
 
-	char *dbName = r_str_newf (R_JOIN_2_PATHS ("syscall", "%s-%s-%d"),
-		os, arch, bits);
-	s->db = openDatabase (s->db, dbName);
-	free (dbName);
+	if (syscall_changed) {
+		char *dbName = r_str_newf (R_JOIN_2_PATHS ("syscall", "%s-%s-%d"),
+			os, arch, bits);
+		if (dbName) {
+			s->db = openDatabase (s->db, dbName);
+			free (dbName);
+		} else {
```

Comment:
Remove this eprintf. We should put an assert line in r_str_newf instead of printing an error in every single place that something can fail. Just keep the if to avoid crashes when the heap is under pressure or buggy

Reply:
No.... assert are not meant to handle runtime errors. Removing this print just means that in case something bad happens you won’t know what went bad.